### PR TITLE
feature: Copy all email alerts to default contact

### PR DESCRIPTION
**Pull request by FTBZ**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

---------------------------------------------------

- Add a way to copy all email alerts to default contact without ignoring other methods.

screen shot 2018-01-31 at 15 09 41

_Thanks @laf for the help._

### CODE REVIEW

File: includes/alerts.inc.php [hunk 1]
```diff
@@ -317,6 +317,11 @@ function GetContacts($results)
         }
     }
 
+    # Copy all email alerts to default contact if configured.
+    if ($config['alert']['default_copy'] === true) {
+        $tmp_contacts[$config['alert']['default_mail']] = '';
```
Comment (laf) [inline includes/alerts.inc.php]: Before assigning it to $tmp_contacts, you need to check it doesn't exist already :)

File: sql-schema/233.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+INSERT INTO `config` VALUES (798,'alert.default_copy','true','true','Copy all email alerts to default contact','alerting',0,'general',0,'0','0');
```
Comment (laf) [inline sql-schema/233.sql]: This is breaking travis, you shouldn't insert the entry with a config_id set as they are auto incrementing.

See sql-schema/221.sql on how to do it.

File: includes/alerts.inc.php [hunk 3]
```diff
@@ -317,6 +317,11 @@ function GetContacts($results)
         }
     }
 
+    # Copy all email alerts to default contact if configured.
+    if (!isset($tmp_contacts[$config['alert']['default_mail']]) && ($config['alert']['default_copy'])) {
+	        $tmp_contacts[$config['alert']['default_mail']] = '';
```
Comment (laf) [inline includes/alerts.inc.php]: This has a tab in from what I can see, needs replacing for 4 spaces :)

Comment (FTBZ) [reply includes/alerts.inc.php]: Oops, but I can't commit anymore: "error: failed to push some refs"

Comment (laf) [reply includes/alerts.inc.php]: You might need to git pull first.

**Pull request merged**